### DISCUSSION TITLE
metaMask

### DISCUSSION THREAD

**Discussion by Amazing123511** [WalletConnect/walletconnect-monorepo#2795] [Developer Support]
Now use the EthereumProvider.init method, the little fox does not respond after scanning the code

**Comment by Amazing123511** [comment]
try {
    const provider = await EthereumProvider.init({
      projectId: projectId, // REQUIRED your projectId
      chains: [1], // REQUIRED chain ids
      optionalChains: [5], // chains added to optional namespaces
      showQrModal: true, // REQUIRED set to "true" to use @walletconnect/modal,
      methods: ["eth_accounts", "eth_requestAccounts", "eth_sign", "personal_sign", "eth_sendTransaction", "eth_chainId", "wallet_switchEthereumChain",  "wallet_addEthereumChain"], // OPTIONAL ethereum methods
      events: [
        "chainChanged",
        "accountsChanged",
        "connect",
        "session_event",
        "display_uri",
        "disconnect",
      ], // OPTIONAL ethereum events
    })
    console.log(provider);
    return provider
  } catch (error) {
    console.log(error);
    openErr(error)
  }

**Reply by filipesoccol** [reply]
Same error reported here. I'm receiving the following error:
Error: Unsupported chains.

Here is my config for them:
 chains: [1, 137, 5, 80001],
 qrModalOptions: {
            chains: [
              'eip155:1',
              'eip155:5',
              'eip155:137',
              'eip155:80001',
            ]
}

Any issue with the config? 

Trust wallet connecting fine with same config.
<img width="435" alt="Screenshot 2023-06-27 at 10 23 17" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/13040410/b4bbf98a-8cca-4edc-ae9c-3b4e50415340">

**Reply by filipesoccol** [reply]
Update on this... If I remove all other options for the chains and keep only Ethereum chain it works. How should I configure it to allow other connections to support other chains?

**Reply by finessevanes** [reply]
You want to add it under optional chains: https://docs.walletconnect.com/2.0/web/providers/ethereum#initialization

**Comment by BitHighlander** [comment]
metamask and uniswap seems broke on wallet-connect v2. it just crashs after scanning.
